### PR TITLE
Fix Rust indentation when a string literal contains "if"

### PR DESCRIPTION
**Pull request by taylordotfish**
indent/rust.vim behaves incorrectly when a string literal contains the substring "if".

For example, in this code:

```rust
let x = "
            motif
";
struct X {
            }
```

indent/rust.vim thinks that the closing "}" should line up with "motif".

This patch fixes the issue by checking whether the "if" is in a string literal or comment before considering it to be a match for a subsequent brace (and also by requiring it to start on a word boundary).

### CODE REVIEW

**Comment by chrisbra** [comment]
thanks. I am a bit worried, this may result in an endless loop. Can you add a simple counter as abort condition? I also have tentatively added a rust indent test like this:

Can you please include it here in this PR together with the additional abort condition:

```diff
diff --git a/runtime/indent/testdir/rust.in b/runtime/indent/testdir/rust.in
index 28a68924f..43a883c06 100644
--- a/runtime/indent/testdir/rust.in
+++ b/runtime/indent/testdir/rust.in
@@ -40,3 +40,11 @@ fn main() {
     // file goes out of scope, and the "hello.txt" file gets closed
 }
 // END_INDENT
+
+// START_INDENT
+let x = "
+            motif
+            ";
+            struct X {
+                        }
+// END_INDENT
diff --git a/runtime/indent/testdir/rust.ok b/runtime/indent/testdir/rust.ok
index 04808e224..4c0145250 100644
--- a/runtime/indent/testdir/rust.ok
+++ b/runtime/indent/testdir/rust.ok
@@ -40,3 +40,11 @@ fn main() {
     // file goes out of scope, and the "hello.txt" file gets closed
 }
 // END_INDENT
+
+// START_INDENT
+let x = "
+            motif
+            ";
+            struct X {
+}
+// END_INDENT
```

You can test it locally by going into `runtime/indent` and running `make in there`

**Comment by taylordotfish** [comment]
Ok, I made the requested changes. I slightly adjusted the test so that it also tests when "if" starts on a word boundary, as well as "fn". Otherwise, just the fact that I changed `\<\(fn\)\|\(if\)\>` to `\<\(fn\|if\)\>` would be enough to make the test pass, without actually testing the loop logic.

**Comment by chrisbra** [comment]
thanks
